crates/pop-cli/src/commands/install/mod.rs

```diff
@@ -0,0 +1,338 @@
+	Libssl,
+	#[strum(serialize = "make")]
+	Make,
+	#[strum(serialize = "openssl")]
+	Openssl,
+	#[strum(serialize = "openssl-devel")]
+	OpenSslDevel,
+	#[strum(serialize = "pkg-config")]
+	PkgConfig,
+	#[strum(serialize = "protobuf")]
+	Protobuf,
+	#[strum(serialize = "protobuf-compiler")]
+	ProtobufCompiler,
+	#[strum(serialize = "rustup")]
+	Rustup,
+}
+
+#[derive(Args)]
+#[command(args_conflicts_with_subcommands = true)]
+/// Setup user environment for development
+pub(crate) struct InstallArgs {
+	/// Automatically install all dependencies required without prompting for confirmation.
+	#[clap(short('y'), long)]
+	skip_confirm: bool,
+}
+
+impl InstallArgs {
+	pub(crate) async fn execute(self) -> anyhow::Result<()> {
+		clear_screen()?;
+		set_theme(Theme);
+		intro(format!(
+			"{}: Install dependencies for development",
+			style(" Pop CLI ").black().on_magenta()
+		))?;
+		if cfg!(target_os = "macos") {
+			log::info("ℹ️ Mac OS (Darwin) detected.")?;
+			install_mac(self.skip_confirm).await?;
+		} else if cfg!(target_os = "linux") {
+			match os_info::get().os_type() {
+				Type::Arch => {
+					log::info("ℹ️ Arch Linux detected.")?;
+					install_arch(self.skip_confirm).await?;
+				},
+				Type::Debian => {
+					log::info("ℹ️ Debian Linux detected.")?;
+					install_debian(self.skip_confirm).await?;
+				},
+				Type::Redhat => {
+					log::info("ℹ️ Redhat Linux detected.")?;
+					install_redhat(self.skip_confirm).await?;
+				},
+				Type::Ubuntu => {
+					log::info("ℹ️ Ubuntu detected.")?;
+					install_ubuntu(self.skip_confirm).await?;
+				},
+				_ => return not_supported_message(),
+			}
+		} else {
+			return not_supported_message();
+		}
+		install_rustup().await?;
+		outro("✅ Installation complete.")?;
+		Ok(())
+	}
+}
+
+async fn install_mac(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/macos/")?;
+	if !skip_confirm {
+		prompt_for_confirmation(&format!(
+			"{}, {}, {}, {} and {}",
+			Dependencies::Homebrew,
+			Dependencies::Protobuf,
+			Dependencies::Openssl,
+			Dependencies::Rustup,
+			Dependencies::Cmake,
+		))?
+	}
+	install_homebrew().await?;
+	cmd("brew", vec!["update"]).run()?;
+	cmd(
+		"brew",
+		vec![
+			"install",
+			&Dependencies::Protobuf.to_string(),
+			&Dependencies::Openssl.to_string(),
+			&Dependencies::Cmake.to_string(),
+		],
+	)
+	.run()?;
+
+	Ok(())
+}
+
+async fn install_arch(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation(&format!(
+			"{}, {}, {}, {}, {} and {}",
+			Dependencies::Curl,
+			Dependencies::Git,
+			Dependencies::Clang,
+			Dependencies::Make,
+			Dependencies::Openssl,
+			Dependencies::Rustup,
+		))?
+	}
+	cmd("pacman", vec!["-Syu", "--needed", "--noconfirm", &Dependencies::Openssl.to_string()])
```

Comment:
Yes. It took me a while to figure out why the CI was failing in `arch` and is because it was missing it.
For the `openSSL` is need in `macos` and `arch`, but for example in `debian` ad `ubuntu` we are installing` libssl-dev`.

You can see the whole list here: https://stackoverflow.com/a/77978063/5964038

Reply:
Can this not just be added with the dependencies on lines 147-150? If it needs to be explicitly installed first, I suggest adding it above Curl first (along with a comment as to why) and if it is explicitly needed to be run as a separate command then it should have a comment noting why.

Reply:
Yes, it can. Merged with the rest of the installs.